sdk/core/azure-core/inc/azure/core/uuid.hpp

```diff
@@ -54,6 +62,60 @@ namespace Azure { namespace Core {
      * @brief Construct a Uuid from an existing UUID represented as an array of bytes.
      * @details Creates a Uuid from a UUID created in an external scope.
      */
-    static Uuid CreateFromArray(std::array<uint8_t, UuidSize> const& uuid);
+    static constexpr Uuid CreateFromArray(ValueArray const& uuid) { return Uuid{uuid}; }
+
+    /**
+     * @brief Construct a Uuid by parsing its representation.
+     * @param s a string in `8-4-4-4-12` hex characters format.
+     * @throw `std::invalid_argument` if \p s cannot be parsed.
+     */
+    static Uuid Parse(std::string const& s);
```

Comment:
We should avoid single letters/abbreviated parameter names that are not descriptive.
`uuidString`?

Reply:
Thanks! See #5943.